graf2d/gpadv7/src/RColor.cxx

```diff
@@ -232,10 +232,10 @@ std::vector<uint8_t> RColor::AsRGBA() const
 
 std::string RColor::toHex(uint8_t v)
 {
-   static const char *digits = "0123456789ABCDEF";
+   auto digits = [](auto d) { return d < 10 ? '0' + d : 'A' - 10 + d; };
```

Comment:
The problem is that there are a lot of static variables/arrays of type RColor that get initialized when libraries are loaded.
When these static variables are initialized during static initialization, the RColor constructor is called, which calls RColor::toHex to initialize one of its members.
However, RColor::toHex has a static variable digits that is also initialized during the static initialization.
And since the order of static initialization is undefined, the initialization of the RColor static variables, that calls RColor::toHex can happen before the digits static variable in RColor::toHex is initialized, and you get a segmentation fault.

Reply:
> And since the order of static initialization is undefined

That's only true for global statics; function statics must be initialized on first use. I.e. you're saying there's a compiler bug somewhere... For ppc64le you're still using GCC I suppose?

Reply:
Yes.